src/components/layout/data/hooks.js

```diff
@@ -0,0 +1,97 @@
+import { useMemo } from 'react';
+import Color from 'color';
+
+import { isDefinedAndNotNull, isDefined } from '../../../utils/common';
+
+const COLOR_LIGHTEN_DARKEN_MODIFIER = 0.2;
+const COLOR_MIX_MODIFIER = 0.1;
+
+// eslint-disable-next-line import/prefer-default-export
+export const useStylesForCustomBrandColors = (enterpriseConfig) => {
+  const brandColors = useMemo(
+    () => {
+      if (isDefinedAndNotNull(enterpriseConfig)) {
+        const { branding } = enterpriseConfig;
+
+        const primaryColor = Color(branding.colors.primary);
+        const secondaryColor = Color(branding.colors.secondary);
+        const tertiaryColor = Color(branding.colors.tertiary);
+
+        const whiteColor = Color('#ffffff');
+        const darkColor = Color('#171c29');
+
+        const getA11yTextColor = color => (color.isDark() ? whiteColor : darkColor);
+
+        return {
+          white: whiteColor,
+          dark: darkColor,
+          primary: {
+            regular: primaryColor,
+            light: primaryColor.lighten(COLOR_LIGHTEN_DARKEN_MODIFIER),
+            dark: primaryColor.darken(COLOR_LIGHTEN_DARKEN_MODIFIER),
+            textColor: getA11yTextColor(primaryColor),
+          },
+          secondary: {
+            regular: secondaryColor,
+            light: secondaryColor.lighten(COLOR_LIGHTEN_DARKEN_MODIFIER),
+            dark: secondaryColor.darken(COLOR_LIGHTEN_DARKEN_MODIFIER),
+            textColor: getA11yTextColor(secondaryColor),
+          },
+          tertiary: {
+            regular: tertiaryColor,
+            light: tertiaryColor.lighten(COLOR_LIGHTEN_DARKEN_MODIFIER),
+            dark: tertiaryColor.darken(COLOR_LIGHTEN_DARKEN_MODIFIER),
+            textColor: getA11yTextColor(tertiaryColor),
+          },
+        };
+      }
+      return undefined;
+    },
+    [enterpriseConfig],
+  );
+
+  if (!isDefined(brandColors)) {
+    return null;
+  }
+
+  const colors = ['primary', 'secondary', 'tertiary'];
+  const styles = colors.map((colorName) => (
+    `
+      .btn-brand-${colorName} {
+        background-color: ${brandColors[colorName].regular.hex()} !important;
```

Comment:
So many `!important` feels not great, but if we're making sure these are always set I get it

Reply:
Yeah, `!important` is typically gross. But, that's how most CSS frameworks work. Like Bootstrap, all their styles include `!important` so I was following that convention here.